### ISSUE TITLE
Network optimisation not taking other hyper-parameters into account, amongst other enhancements

### ISSUE THREAD

**Issue by olliekemp** [reac2/Tiny#4]
The network builders in ``network_builder.py`` only tune over the number of layers, nodes in each layer and learning rate. There are a lot of other hyper parameters we can optimise. It would also be good if we can implement some more advanced training techniques - learning rate decay, early stopping etc. Finally, we _could_  use the parallel evaluation stuff in [optuna](https://optuna.org) to speed up the evaluation on some proper compute but I haven't looked into that very much.
